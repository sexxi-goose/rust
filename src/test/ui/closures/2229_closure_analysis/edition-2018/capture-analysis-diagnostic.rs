// edition:2018
#![feature(rustc_attrs)]

struct S(i32);

fn foo() {
    let b = Box::new(S(10));

    let c = #[rustc_capture_analysis]
    //~^ ERROR: attributes on expressions are experimental
    //~| NOTE: see issue #15701 <https://github.com/rust-lang/rust/issues/15701>
    || {
    //~^ First Pass analysis includes:
    //~| Min Capture analysis includes:
        let _b = b.0;
        //~^ NOTE: 15:18: 15:21: Min Capture b[] -> ImmBorrow
        //~| NOTE: 15:18: 15:21: Capturing b[Deref,(0, 0)] -> ImmBorrow
    };

    c();
}

fn main() {
    foo();
}
