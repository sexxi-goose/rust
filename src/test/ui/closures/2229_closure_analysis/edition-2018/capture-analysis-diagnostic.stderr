error[E0658]: attributes on expressions are experimental
  --> $DIR/capture-analysis-diagnostic.rs:9:13
   |
LL |     let c = #[rustc_capture_analysis]
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #15701 <https://github.com/rust-lang/rust/issues/15701> for more information
   = help: add `#![feature(stmt_expr_attributes)]` to the crate attributes to enable

error: First Pass analysis includes:
  --> $DIR/capture-analysis-diagnostic.rs:12:5
   |
LL | /     || {
LL | |
LL | |
LL | |         let _b = b.0;
LL | |
LL | |
LL | |     };
   | |_____^
   |
note: Capturing b[Deref,(0, 0)] -> ImmBorrow
  --> $DIR/capture-analysis-diagnostic.rs:15:18
   |
LL |         let _b = b.0;
   |                  ^^^

error: Min Capture analysis includes:
  --> $DIR/capture-analysis-diagnostic.rs:12:5
   |
LL | /     || {
LL | |
LL | |
LL | |         let _b = b.0;
LL | |
LL | |
LL | |     };
   | |_____^
   |
note: Min Capture b[] -> ImmBorrow
  --> $DIR/capture-analysis-diagnostic.rs:15:18
   |
LL |         let _b = b.0;
   |                  ^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0658`.
